testsuite/MDAnalysisTests/analysis/test_nucleicacids.py

```diff
@@ -41,5 +41,12 @@ def test_wc_dist(u):
     WC = WatsonCrickDist(strand1, strand2)
     WC.run()
 
-    assert_allclose(WC.results[0][0], 4.3874702, atol=1e-3)
-    assert_allclose(WC.results[1][0], 4.1716404, atol=1e-3)
+    assert_allclose(WC.results.pair_distances[0, 0], 4.3874702, atol=1e-3)
+    assert_allclose(WC.results.pair_distances[0, 1], 4.1716404, atol=1e-3)
+
+    wmsg = ("Accessing results via selection indices is "
+            "deprecated and will be removed in MDAnalysis 2.5.0")
+
+    with pytest.warns(match=wmsg):
```

Comment:
MAybe use https://docs.pytest.org/en/6.2.x/reference.html#pytest.deprecated_call
```python
with pytest.deprecated_call(match=wmsg):
```

Reply:
That's a new one to me, sounds fun.

Good side-effect catch, I'm emitting the wrong warning type too.